scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -313,23 +312,37 @@ class ScioContext private[scio] (val options: PipelineOptions,
   private var _pipeline: Pipeline = _
   private var _isClosed: Boolean = false
   private val _promises: MBuffer[(Promise[Tap[_]], Tap[_])] = MBuffer.empty
-  private val _queryJobs: MBuffer[QueryJob] = MBuffer.empty
   private val _preRunFns: MBuffer[() => Unit] = MBuffer.empty
   private val _counters: MBuffer[Counter] = MBuffer.empty
+  private var _onClose: Unit => Unit = identity
+  private val _localInstancesCache: scala.collection.mutable.Map[ClassTag[_], Any] =
```

Comment:
I'm not super happy with this...

 The bigquery client instance was a local val inside ScioContext which guaranted it to be unique per context. Using implicit conversions has the consequence of instantiating a new client every time `ScioContext` is converted to `BigQueryScioContext`. This thing is used to make sure we still only instantiate one client.

Reply:
Where is the new client initiated in `ScioContext` now? `private lazy val bigQueryClient` is removed.

Reply:
It's initiated in BigQueryIO.scala